plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -66,9 +66,11 @@ def set_deployment_metadata
       end
     end
 
+    # labels that are used to match the previous deployment,
+    # they cannot change or we will create a new deployment instead of updating the previous one
+    # re-creating roles or re-creating the deploy group will lead to duplicate deployments
     def deployment_labels
-      # Deployment labels should not include the ids of the release, role or deploy groups
-      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
+      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
```

Comment:
ah perfect, so I'll keep this PR as it is and make a new one that no longer modifies deployment and selector and only adds (never overrides) labels to pods -> yml file must be valid / we will never make it invalid :D